pkg/domain/infra/abi/containers.go

```diff
@@ -1131,6 +1131,10 @@ func (ic *ContainerEngine) ContainerRun(ctx context.Context, opts entities.Conta
 		}
 	}
 
+	if opts.Rm && opts.Spec.SdNotifyMode == define.SdNotifyModeHealthy {
+		return nil, fmt.Errorf("cannot use together --rm and --sdnotify=healthy")
```

Comment:
suggestion: `cannot use --rm with --sdnotify=healthy`

(This is a language note only, and not meant as endorsement of your approach in this PR. I find this a little uncomfortable, like just sweeping the problem under the carpet. But I'll let others comment on that aspect).

Reply:
This is a breaking change and should not be done. It definitely breaks quadlets that use `Notify=healthy` as they use --rm by default.

Preventing the container from being create/started is a big change, I understand that sdnotify=healthy does not really work with short living containers due the races but it cannot be changed.

I would suggest for `podman run` exit as normal and ignore sdnotify=health errors if we cannot read the healtcheck. The sdnotify listener will know it never got ready and the mainpid exited so that seems fine and no reason to make the `podman run` fail because of it. I would just documented that possibility.

Reply:
this was a rushed fix and I've left it marked as a Draft as I was not happy with it either, I've opened the PR early just to run the CI for the other patch.

When `--sdnotify=healthy` is used we need to unlock the container to give the "healthcheck" podman process the possibility to run and access the container.  Doing so, also gives the "cleanup" process the possibility to run.

When that happens and `--rm` is provided, the cleanup process rightfully removes all the containers storage, including the container exit status.  So we cannot retrieve the exit status once we get the lock back.

This doesn't happen if `--sdnotify=healthy` is used, since the podman process holds the container lock all the time, and the cleanup process cannot delete the information we need.

I've proposed an alternative fix now.  It disables the auto removal from the cleanup process.  This is not ideal either, but at least we can retrieve the exit status now and `--sdnotify=healthy --rm` can be used.

Reply:
> When that happens and --rm is provided, the cleanup process rightfully removes all the containers storage, including the container exit status. So we cannot retrieve the exit status once we get the lock back.

This is not true, we store the exit code in a separate db bucket/table in order to allow this. This is why a normal `podman run` works, it obviously also unlocks after it started the container and attached stdin/out/err.

Reply:
thanks, that is good to know!  I've changed the patch to use this information from `WaitForExit`